wire-ios/Wire-iOS/Sources/UserInterface/ConversationList/Container/ConversationListViewController+NavigationBar/ConversationListViewController+NavigationBar.swift

```diff
@@ -0,0 +1,401 @@
+        accountImageView.heightAnchor.constraint(equalToConstant: 28).isActive = true
+
+        return accountImageView
+    }
+
+    func setupLeftNavigationBarButtonItems() {
+
+        // in the design the left bar button items are very close to each other,
+        // so we'll use a stack view instead
+        let stackView = UIStackView()
+        stackView.spacing = 4
+
+        // avatar
+        let accountImageView = setupAccountImageView()
+        stackView.addArrangedSubview(accountImageView)
+        self.accountImageView = accountImageView
+
+        // legal hold
+        switch viewModel.selfUserLegalHoldSubject.legalHoldStatus {
+        case .disabled:
+            break
+        case .pending:
+            let pendingRequestView = createPendingLegalHoldRequestView()
+            stackView.addArrangedSubview(pendingRequestView)
+        case .enabled:
+            let legalHoldView = createLegalHoldView()
+            stackView.addArrangedSubview(legalHoldView)
+        }
+
+        // verification status
+        if viewModel.selfUserStatus.isE2EICertified {
+            let imageView = UIImageView(image: .init(resource: .certificateValid))
+            imageView.contentMode = .scaleAspectFit
+            stackView.addArrangedSubview(imageView)
+        }
+        if viewModel.selfUserStatus.isProteusVerified {
+            let imageView = UIImageView(image: .init(resource: .verifiedShield))
+            imageView.contentMode = .scaleAspectFit
+            stackView.addArrangedSubview(imageView)
+        }
+
+        navigationItem.leftBarButtonItems = [.init(customView: stackView)]
+    }
+
+    func setupLeftNavigationBarButtonItems_SplitView() {
+        navigationItem.leftBarButtonItems = []
+    }
+
+    func setupTitleView() {
+        switch mainSplitViewState {
+        case .expanded:
+            switch conversationFilter {
+            case .none:
+                navigationItem.title = L10n.Localizable.ConversationList.Filter.AllConversations.title
+            case .favorites:
+                navigationItem.title = L10n.Localizable.ConversationList.Filter.Favorites.title
+            case .groups:
+                navigationItem.title = L10n.Localizable.ConversationList.Filter.Groups.title
+            case .oneOnOne:
+                navigationItem.title = L10n.Localizable.ConversationList.Filter.OneOnOneConversations.title
+            }
+        case .collapsed:
+            navigationItem.title = L10n.Localizable.List.title
+        }
+    }
+
+    func setupRightNavigationBarButtonItems() {
+
+        let spacer = UIBarButtonItem(systemItem: .fixedSpace)
+        spacer.width = 18
+        typealias FilterMenuLocale = L10n.Localizable.ConversationList.Filter
+
+        // New Conversation Button
+        let symbolConfiguration = UIImage.SymbolConfiguration(weight: .semibold)
+        let newConversationImage = UIImage(systemName: "plus.circle.fill", withConfiguration: symbolConfiguration)!
```

Comment:
I assume these force unwrap would go away when we use Swiftgen constants, right?

Reply:
This is code for creating an SF Symbols image. I don't think SwiftGen will be helpful for this. Maybe I'm wrong.